File: app/src/main/java/com/ps/simplepapersoccer/AI/Abstraction/IGameAI.java
package com.ps.simplepapersoccer.AI.Abstraction;

import com.ps.simplepapersoccer.GameObjects.Game.GameHandler;
import com.ps.simplepapersoccer.GameObjects.Move.PartialMove;

public interface IGameAI
{
    public PartialMove MakeMove(GameHandler gameHandler);
}

File: app/src/main/java/com/ps/simplepapersoccer/ai/abstraction/IGameAI.kt
package com.ps.simplepapersoccer.ai.abstraction

import com.ps.simplepapersoccer.gameobjects.game.GameHandler
import com.ps.simplepapersoccer.gameobjects.move.PartialMove
import java.io.Serializable

interface IGameAI: Serializable {
    suspend fun makeMove(gameHandler: GameHandler): PartialMove?
}
